### PR TITLE
Gzip fat beacon

### PR DESCRIPTION
**Pull request by iankchristie**

### CODE REVIEW

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java [hunk 1]
```diff
@@ -193,9 +198,22 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
     }
 
     characteristic = gatt.getService(SERVICE_UUID).getCharacteristic(CHARACTERISTIC_WEBPAGE_UUID);
+    checkWebpageGzipped();
+    
     gatt.readCharacteristic(characteristic);
   }
 
+  private void checkWebpageGzipped() {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: It'd be nice if these both used the same characteristic, and you just inspected the content at the point when you needed it to see whether it is gzipped.

Comment (iankchristie) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: it uses the same characteristic. The descriptor is nested within the characteristic.
Check the content of the file to see if it is gzipped? Is there an easy way to do that?

Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: OK, then same descriptor I guess.  You could try unzipping it and see if it fails.  If that's the case, it's not gzipped.

Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: See also GZIP_MAGIC here: http://stackoverflow.com/questions/4818468/how-to-check-if-inputstream-is-gzipped

**Comment by iankchristie** [comment]
@cco3 made fix for GZIP_MAGIC. Not using the descriptor anymore

**Comment by cco3** [comment]
Looks OK.  Please squash.

**Comment by iankchristie** [comment]
@cco3 squashed

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java [hunk 2]
```diff
@@ -192,7 +192,7 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
       return;
     }
 
-    characteristic = gatt.getService(SERVICE_UUID).getCharacteristic(CHARACTERISTIC_WEBPAGE_UUID);
+    characteristic = gatt.getService(SERVICE_UUID).getCharacteristic(CHARACTERISTIC_WEBPAGE_UUID);
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: tws

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java [hunk 3]
```diff
@@ -228,7 +228,16 @@ private File getHtmlFile() {
     return new File(websiteDir, "website.html");
   }
 
+  private File getTempFile() {
+    File websiteDir = new File(activity.getFilesDir(), "Websites");
+    websiteDir.mkdir();
+    return new File(websiteDir, "temp.html");
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: See File.createTempFile() http://stackoverflow.com/questions/3425906/creating-temporary-files-in-android

Comment (iankchristie) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: trying to implement the File.createTempFile() and I keep getting error
java.lang.IllegalArgumentException: Failed to find configured root that contains /data/data/physical_web.org.physicalweb/cache/temp-72855774.tmp

Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: Hm, you might need to set a cache-path? https://developer.android.com/reference/android/support/v4/content/FileProvider.html

Comment (iankchristie) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java]: I added this to the xml/file_path.xml
<cache-path name="cache" path="cachepath" />
and i receive the same error
java.lang.IllegalArgumentException: Failed to find configured root that contains /data/data/physical_web.org.physicalweb/cache/temp-798576872.tmp

**Comment by iankchristie** [comment]
@cco3 squashed everything

**Comment by cco3** [comment]
LGTM, but let's have Matt look at it first.

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java [hunk 4]
```diff
@@ -1119,4 +1126,64 @@ public PwsResultBuilder setPwsTripTimeMillis(PwsResult pwsResult, long timeMilli
...
+  public static boolean isGzippedFile(File file) {
+    InputStream input;
+
+    try {
+      input = new FileInputStream(file);
+    } catch(FileNotFoundException e) {
+      return false;
+    }
+
+    PushbackInputStream pb = new PushbackInputStream(input, GZIP_SIGNATURE_LENGTH);
```
Comment (nondebug) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java]: Why PushbackInputStream? Could you use FileInputStream.read instead? It doesn't seem like we need the pushback functionality.

**Comment by iankchristie** [comment]
@cco3 @nondebug hey hey, nice catch Matt. We didn't need that functionality since we aren't changing the underlying file with the stream

**Comment by nondebug** [comment]
lgtm

**Comment by cco3** [comment]
You need to rebase on master.

**Comment by cco3** [comment]
Also, for the commit message, don't use a full stop on the subject line.  Please use full sentences in the message body.

**Comment by cco3** [comment]
You forgot to update the commit message.

**Comment by iankchristie** [comment]
@cco3 how does it look now?

**Pull request merged**